specification.markdown

```diff
@@ -245,6 +245,18 @@ A `Binary Element` MUST declare a length in octets from zero to `VINTMAX`.
 
 The contents of a `Binary Element` should not be interpreted by the `EBML Reader`.
 
+# Terminating Elements
+
+`Null Octets`, which are octets with all bits set to zero, MAY follow the value of a `String Element` or `UTF-8 Element` to serve as a terminator. An `EBML Writer` MAY terminate a `String Element` or `UTF-8 Element` with `Null Octets` in order to overwrite a stored value with a new value of lesser length while maintaining the same `Element Data Size` (this can prevent the need to rewrite large portions of an `EBML Document`); otherwise the use of `Null Octets` within a `String Element` or `UTF-8 Element` is NOT RECOMMENDED. An `EBML Reader` MUST consider the value of the `String Element` or `UTF-8 Element` to be terminated upon the first read `Null Octet` and ignore any following data within that `Element`. If an `EBML Writer` terminates the value of a `String Element` or `UTF-8 Element` with a `Null Octet` then any subsequent octets of the `Element` MUST also be `Null Octets`. A string value and a copy of that string value terminated by one or more `Null Octets` are semantically equal.
```

Comment:
That's what I was thinking about, yeah.

I tried to answer the general question of what our recommendation should be if any element (not necessarily one of the two string types) needs to be extended/shrunk. It can easily happen for master elements.

This leaves us with only one viable strategy that I can see at the moment: re-writing the element located before that one byte hole and extending its vint size.